raft/src/main/java/org/apache/kafka/raft/internals/RecordsIterator.java

```diff
@@ -179,7 +184,12 @@ private Optional<Batch<T>> nextBatch() {
         return Optional.empty();
     }
 
-    private Batch<T> readBatch(DefaultRecordBatch batch) {
+    private Batch<T> readBatch(DefaultRecordBatch batch) throws CorruptRecordException {
```

Comment:
`CorruptRecordException` is a runtime exception and doesn't need to be added to the set of checked exceptions.